src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -44,6 +45,47 @@ for (const funcName of Object.keys(vegaFunctions)) {
 
 const bypassToken = Symbol();
 
+function normalizeDate(date) {
+  let normalizedDate = '';
+  if (typeof date === 'number') {
+    normalizedDate = date;
+  } else if (date instanceof Date) {
+    normalizedDate = date;
+  } else {
+    normalizedDate = normalizeString(date);
+  }
+  return normalizedDate;
+}
+
+// recursive function to check a nested object for the 'toJSON' property
+function checkObjectForToJsonProperty(object) {
```

Comment:
__question:__ since we already recursively walk through the objects, is there any reason why we cannot reject _any_ `typeof === 'function'` property?